app-modules/segment/src/Actions/BulkSegmentAction.php

```diff
@@ -0,0 +1,100 @@
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Advising App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AdvisingApp\Segment\Actions;
+
+use Exception;
+use Filament\Forms\Form;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use AdvisingApp\Segment\Models\Segment;
+use Filament\Forms\Components\Textarea;
+use Filament\Tables\Actions\BulkAction;
+use Filament\Forms\Components\TextInput;
+use Filament\Notifications\Notification;
+use AdvisingApp\Segment\Enums\SegmentType;
+use AdvisingApp\Segment\Enums\SegmentModel;
+
+class BulkSegmentAction
+{
+    public static function make(SegmentModel $segmentModel): BulkAction
+    {
+        return BulkAction::make('segment')
+            ->icon('heroicon-o-rectangle-group')
+            ->label('Create Segment')
+            ->form(fn (Form $form) => $form->schema([
+                TextInput::make('name')
+                    ->label('Name')
+                    ->required()
+                    ->maxLength(255),
+
+                Textarea::make('description')
+                    ->label('Description')
+                    ->maxLength(65535),
+            ]))
+            ->action(function (Collection $records, array $data) use ($segmentModel) {
+                try {
+                    DB::beginTransaction();
+                    $data['type'] = SegmentType::Static;
+                    $data['filters'] = [];
+                    $data['model'] = $segmentModel;
+                    $segment = Segment::create($data);
+                    $records->chunk(100)->each(function ($chunkRecord) use ($segment) {
+                        $subjectData = $chunkRecord->map(fn ($record) => [
+                            'subject_id' => $record->getKey(),
+                            'subject_type' => $record->getMorphClass(),
+                        ])->toArray();
+                        $segment->subjects()->createMany($subjectData);
+                    });
+                    DB::commit();
```

Comment:
A few notes, I would really reccomend giving your code some logical breathing room. I can really help with the organization and readability.

Also, since you are immediately defining the `$subjectData` and using it, it could just be done inline.

All like so:

```suggestion
                    DB::beginTransaction();
                    
                    $segment = Segment::create([
                        ...$data,
                        'type' => SegmentType::Static,
                        'filters' => [],
                        'model' => $segmentModel,
                    ]);
                    
                    $records->chunk(100)->each(fn ($chunkRecord) => $segment
                        ->subjects()
                        ->createMany(
                            $chunkRecord->map(fn ($record) => [
                                'subject_id' => $record->getKey(),
                                'subject_type' => $record->getMorphClass(),
                            ])->toArray()
                        )
                    );
                    
                    DB::commit();
```

I did this in the GitHub comment input so hopefully that is all correct lol. You will want to confirm and test.

Reply:
Yeh, @Orrison thanks for showing me the inline changes, I'll keep this in mind, Your code is working fine even you wrote in the comment.